src/filter/segfilter.hpp

```diff
@@ -1,24 +1,23 @@
 #ifndef SEG_FILTER_H
 #define SEG_FILTER_H
-
 #include "IFilter.hpp"
+#include "IModel.hpp"
 #ifdef USE_MOCK
 #include "../hal_mock/model/MockTRT.hpp"
 #else
 #include "../hal_jetson/model/TensorRTModel.hpp"
 #endif
-
-
 #include <stdio.h>
 #include "../utils/ThreadLogger.hpp"
 
+
 class SegFilter: public IFilter{
 private:
     /* data */
-    IModel * model;
+    std::vector<IModel*> models;
     ThreadLogger * logger;
 public:
-    SegFilter(const char *modelPath,ThreadLogger * logger);
+    SegFilter(const std::vector<std::string>& modelPaths, ThreadLogger* logger);
```

Comment:
I am not entirely sure if this is the best way to do as now the pipeline can't execute the models on request.